### PR TITLE
[flutter_local_notifications] added ability to specify silent notification through AndroidNotificationDetails

### PR DESCRIPTION
**Pull request by aa-euclidk**
Thanks to [TheFinestArtist](https://github.com/TheFinestArtist) for providing code for adding the silent option

As this repository hosts multiple packages, please ensure the PR title starts with the name of the package that it relates to using square brackets (e.g. [flutter_local_notifications]). Should the PR more than one package than please prefix the title of the PR with [various] The contribution guidelines can be found at https://github.com/MaikuB/flutter_local_notifications/blob/master/CONTRIBUTING.md. Please review this as it contains details on what to follow when submitting a PR.

### CODE REVIEW

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/NotificationDetails.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public class NotificationDetails implements Serializable {
   private static final String ONGOING = "ongoing";
   private static final String STYLE = "style";
   private static final String ICON = "icon";
+  private static final String SILENT = "silent";
```
Comment (TheFinestArtist) [inline flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/NotificationDetails.java:39]: Nit: you might want to write this below `ONGOING` to keep the order with the rest

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java [hunk 2]
```diff
@@ -277,7 +277,8 @@ protected static Notification createNotification(
             .setContentIntent(pendingIntent)
             .setPriority(notificationDetails.priority)
             .setOngoing(BooleanUtils.getValue(notificationDetails.ongoing))
-            .setOnlyAlertOnce(BooleanUtils.getValue(notificationDetails.onlyAlertOnce));
+            .setOnlyAlertOnce(BooleanUtils.getValue(notificationDetails.onlyAlertOnce))
+            .setSilent(BooleanUtils.getValue(notificationDetails.silent));
```
Comment (TheFinestArtist) [inline flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java:281]: Nit: you might want to write this below `ONGOING` to keep the order with the rest

File: flutter_local_notifications/example/ios/Runner.xcodeproj/project.pbxproj
Comment (TheFinestArtist) [inline flutter_local_notifications/example/ios/Runner.xcodeproj/project.pbxproj:1]: I am not super familiar, so I might not be right. But, I don't think this file change is related to the feature.

File: flutter_local_notifications/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme
Comment (TheFinestArtist) [inline flutter_local_notifications/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:1]: I am not super familiar, so I might not be right. But, I don't think this file change is related to the feature.

File: flutter_local_notifications/test/android_flutter_local_notifications_test.dart
Comment (TheFinestArtist) [inline flutter_local_notifications/test/android_flutter_local_notifications_test.dart:1]: Test looks good to me :D

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/NotificationDetails.java (hunk 1 above)
Comment (aa-euclidk) [reply flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/NotificationDetails.java]: Fixed :)

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java (hunk 2 above)
Comment (aa-euclidk) [reply flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java]: Fixed :)

File: flutter_local_notifications/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme
Comment (aa-euclidk) [reply flutter_local_notifications/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:1]: I agree but I'm not entirely sure. Feel free to remove it.

File: flutter_local_notifications/example/ios/Runner.xcodeproj/project.pbxproj
Comment (aa-euclidk) [reply flutter_local_notifications/example/ios/Runner.xcodeproj/project.pbxproj:1]: I agree but I'm not entirely sure. Feel free to remove it.

**Pull request merged**